scala_proto/scala_proto_toolchain.bzl

```diff
@@ -0,0 +1,67 @@
+load("//scala_proto/private:dep_sets.bzl", "SCALAPB_DEPS", "GRPC_DEPS")
+
+def _scala_proto_toolchain_impl(ctx):
+    toolchain = platform_common.ToolchainInfo(
+        with_grpc = ctx.attr.with_grpc,
+        with_flat_package = ctx.attr.with_flat_package,
+        with_single_line_to_string = ctx.attr.with_single_line_to_string,
+        blacklisted_protos = ctx.attr.blacklisted_protos,
+        code_generator = ctx.executable.code_generator,
+        override_code_generator = ctx.executable.override_code_generator,
+        override_code_generator_targets = ctx.attr.override_code_generator_targets,
+        grpc_deps=ctx.attr.grpc_deps,
+        implicit_compile_deps=ctx.attr.implicit_compile_deps,
+        scalac=ctx.attr.scalac,
+    )
+    return [toolchain]
+
+
+# Args:
+#     with_grpc: Enables generation of grpc service bindings for services
+#     with_flat_package: When true, ScalaPB will not append the protofile base name to the package name
+#     with_single_line_to_string: Enables generation of toString() methods that use the single line format
+#     blacklisted_protos: list of protobuf targets to exclude from recursive building
+#     code_generator: what code generator to use, usually you'll want the default
+scala_proto_toolchain = rule(
+    _scala_proto_toolchain_impl,
+    attrs = {
+        "with_grpc": attr.bool(),
+        "with_flat_package": attr.bool(),
+        "with_single_line_to_string": attr.bool(),
+        "blacklisted_protos": attr.label_list(default=[]),
+        "code_generator": attr.label(
```

Comment:
So after some `bazel clean` action this seems to be a problem -- if i have the code generator here, and its an executable dependency. The runfiles aren't available to the actions generated in the aspect. If i add it as a dep to the aspect it works. Anyone know if there is some syntax i'm missing here to thread a proper dependency onto the target via the tool chain on the generator code?

Reply:
this is resolved by 0.23, right?

Reply:
Yep